src/plugins/intel_cpu/src/nodes/kernels/jit_eltwise_common.cpp

```diff
@@ -69,27 +69,46 @@ ov::element::Type eltwise_precision_helper::get_precision(const size_t inputs_nu
         supported_precision_intersection = prcs_intersect;
     }
 
+    // To select the most suitable precision from inputs are mixed-precision
+    // Preference is given to higher bitwidth, and for equal bitwidth, to real (floating point) types.
+    auto select_precision_from_inputs = [](const ov::element::Type(&src_prc)[MAX_ELTWISE_INPUTS],
+                                           const size_t inputs_number) {
+        auto selected_type = src_prc[0];  // Start with the first input's precision
+        for (size_t i = 1; i < inputs_number; i++) {
+            if (selected_type.bitwidth() > src_prc[i].bitwidth()) {
+                continue;
+            }
+            // If bitwidths are equal and selected_type is real (floating point), keep it
+            if (selected_type.bitwidth() == src_prc[i].bitwidth() && selected_type.is_real()) {
+                continue;
+            }
+            // Otherwise, update selected_type to the current input's precision
+            selected_type = src_prc[i];
+        }
+        return selected_type;
+    };
+
+    const auto input_precision = select_precision_from_inputs(src_prc, inputs_number);
```

Comment:
```suggestion
    const auto input_precision = [&] {
        auto selected_type = src_prc[0];  // Start with the first input's precision
        for (size_t i = 1; i < inputs_number; i++) {
            if (selected_type.bitwidth() > src_prc[i].bitwidth()) {
                continue;
            }
            // If bitwidths are equal and selected_type is real (floating point), keep it
            if (selected_type.bitwidth() == src_prc[i].bitwidth() && selected_type.is_real()) {
                continue;
            }
            // Otherwise, update selected_type to the current input's precision
            selected_type = src_prc[i];
        }
        return selected_type;
    } ();
```